src/metaopt/io/database/mongodb.py

```diff
@@ -19,95 +20,121 @@
     from urllib import quote_plus
 
 
+
 class MongoDB(AbstractDB):
     def __init__(self, dbhost='localhost', dbname='metaopt', dbport=27017, \
                                             username=None, password=None):
         self.dbhost = str(dbhost)
         self.dbname = str(dbname)
         self.username = str(username)
         self.password = str(password)
-
-        # we might have a different port for the database port
-        # so here we do a check if the port is there
-
         self.dbport = str(dbport)
 
         # set the connection string
         # this should be mongodb://mongodb0.example.net:27017
         # according to MongoDB documentation
         # https://docs.mongodb.com/getting-started/python/client/
 
-        self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
-        #self.uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self.username), \
-        #                                      quote_plus(self.password), \
-        #                                     self.dbhost, self.dbport)
         # try to initialize connection
         # setting this as a function allows to make sure reconnection
         # is properly done in the case of an unreliable network
 
         self.conn = self.initiate_connection()
         self.db = self.conn[self.dbname]
 
+        if __debug__:
+            self.logger = logging.getLogger('database')
+            self.fh = logging.FileHandler('database.log')
+            self.fh.setLevel(logging.DEBUG)
```

Comment:
I understand why we would want a detailed log for the database but do we really want to have DEBUG by default? I guess that's like system's log that are always very verbose by default.

Reply:
Doesn't the `if __debug__:` block only run if we run the script with the flag `python -O abc.py`? So it wouldn't run by default?